src/header/types/accept.rs

```diff
@@ -0,0 +1,325 @@
+    /// ```
+    pub fn as_str(&self) -> String {
+        let m_type = self.m_type.to_string();
+        match &self.quality {
+            Some(quality) => format!("{} ;{}", m_type, quality.as_str()),
+            None => format!("{}", m_type)
+        }
+    }
+}
+
+impl<'accept> TryFrom<&'accept [u8]> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'accept [u8]) -> Result<Self, Self::Error> {
+        let mut split = value.split(|element| element.clone() == ';' as u8);
+        let mut decoded = MediaType::default();
+        if let Some(mediatype) = split.next() {
+            if let Ok(mtype) = str::from_utf8(mediatype).unwrap().parse() {
+                decoded.m_type = mtype;
+                if let Some(quality) = split.next() {
+                    if let Ok(quality_param) = QualityParam::try_from(quality) {
+                        decoded.quality = Some(quality_param);
+                    } else {
+                        return Err(AcceptError("unable to parse quality parameter"))
+                    }
+                } 
+            } else{
+                return Err(AcceptError("unable to parse mime type"))
+            }
+        }
+        Ok(decoded)
+    }
+}
+
+impl<'accept> TryFrom<&'accept str> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'accept str) -> Result<Self, Self::Error> {
+        let bytes = value.as_bytes();
+        Self::try_from(bytes)
+    } 
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct QualityParam(f32);
+
+impl<'quality> TryFrom<&'quality [u8]> for QualityParam {
+    type Error = AcceptError;
+
+    fn try_from(value: &'quality [u8]) -> Result<Self, Self::Error> {
+        let stringify = String::from_utf8(value.to_vec()).unwrap();
+        let val = stringify.split("=").collect::<Vec<&str>>();
+        if val.len() < 2 || val.len() > 2 {
+            return Err(AcceptError("incorrect quality format"))
+        }        
+        let quality = val[1].parse::<f32>();
+        match quality {
+            Ok(quality_param) =>  return Ok(QualityParam::new(quality_param)),
+            Err(_) => return Err(AcceptError("unable to parse f32 from string"))
+        }
+    }
+}
+
+impl<'quality> TryFrom<&'quality str> for QualityParam {
+    type Error = AcceptError;
+
+    fn try_from(value: &'quality str) -> Result<Self, Self::Error> {
+        let bytes = value.as_bytes();
+        Self::try_from(bytes)
+    }
+}
+
+impl QualityParam {
+
+    pub fn new(q_value: f32) -> Self {
+        let q_value = q_value.clamp(0.0_f32, 1.0_f32);
+        QualityParam(q_value)
+    }
+
+
+    /// Returns a `String` representation of the quality parameter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::types::accept::QualityParam;
+    ///
+    /// assert_eq!(QualityParam::new(0.5).as_str(), "q=0.5");
+    /// ```
+    pub fn as_str(&self) -> String {
```

Comment:
Same here, if you can't do `as_str` without allocating, then it shouldn't be used.